### PR TITLE
Update state with API POST & GET hitting functions

### PR DESCRIPTION
**Pull request by MitchellDill**
Config updates done alongside Brandon--if anything here looks suspect, let's talk about it/figure out what our current package/lint/prettier setup should all look like. Actual intent was just to add Nodemon to dev dependencies at the root.

Budget page and Dashboard should be receiving "updateAccountData" function passed down in props, which accepts one argument, the entire user object. This function will post to the DB. UserObject (in state and props as "accountData") modification needs to be done before passing the userObject up.

### CODE REVIEW

**Comment by MitchellDill** [comment]
I'm unsure exactly why this has such a length commit history (and apologize for this appearing to change a bunch of stuff in files when it's actually just prettier/lint onSave flipping quotation marks and parentheses and whatnot in its formatting.

File: client/src/App.jsx [hunk 1]
```diff
@@ -21,47 +19,98 @@ import LoginPage from './Components/LoginPage.jsx';
...
   componentDidMount() {
-    const data = fakeData.createData();
+    this.getUserData('chadchadson@gmail.com')
+      .then(this.setCurrentUser)
+      .then(this.setAccountDataAndBudgetCategories)
+      .catch(err => {
+        console.log('mounting error: ', err);
+      });
+  }
+
```
Comment (MitchellDill) [inline client/src/App.jsx]: The linter wanted ComponentDidMount up here for some reason--any ideas why?

File: client/src/Components/DashboardPage.jsx [hunk 2]
```diff
@@ -289,5 +289,6 @@ DashboardPage.propTypes = {
   accountData: PropTypes.object,
   handleAddTransaction: PropTypes.func,
   loading: PropTypes.bool.isRequired,
+  updateAccountData: PropTypes.func,
```
Comment (MitchellDill) [inline client/src/Components/DashboardPage.jsx]: aside from this, this was just the linter going wild on dashboardPage--shouldn't be any other actual changes

**Comment by rcalimlim** [comment]
Closing because these are breaking changes.

**Pull request closed without merging**